### PR TITLE
Add service objective and operations .NET client tests, upgrade to Az…

### PR DESCRIPTION
**Pull request by nathannfan**
…ure Resource Manager 1.6

## Description

Add .NET tests for service objectives and capabilities.  Record with Azure Resource Manager version 1.6.  

Re-recordings of tests will be done in a separate PR.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. -
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@
 [assembly: AssemblyTitle("Microsoft Azure SQL Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure SQL.")]
 [assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.4.0.0")]
+[assembly: AssemblyFileVersion("1.4.0.1")]
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs]: Is there any product code change, or just test code? If test only, then we don't need to publish so I don't think this needs to be bumped

**Review by jaredmoo — APPROVED** [review]

**Comment by nathannfan** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/3267

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

**Comment by jaredmoo** [comment]
Please review because this is blocking other sql sdk changes :) thx

File: src/SDKs/SqlManagement/Sql.Tests/SessionRecords/Sql.Tests.BlobAuditingTest/TestServerBlobAuditing.json [hunk 2]
```diff
@@ -825,7 +825,7 @@
         ],
         "User-Agent": [
           "FxVersion/4.6.25009.01",
-          "Microsoft.Azure.Management.Resources.ResourceManagementClient/1.0.0-preview"
+          "Microsoft.Azure.Management.ResourceManager.ResourceManagementClient/1.0.0-preview"
```
Comment (shahabhijeet) [inline src/SDKs/SqlManagement/Sql.Tests/SessionRecords/Sql.Tests.BlobAuditingTest/TestServerBlobAuditing.json]: @nathannfan how is this possible, if you are using 1.6.0 version of resource manager with the new namespace, how come you are still recording the old version?

**Review by shahabhijeet — DISMISSED** [review]

File: src/SDKs/SqlManagement/Sql.Tests/SessionRecords/Sql.Tests.BlobAuditingTest/TestServerBlobAuditing.json (hunk 2 above)
Comment (nathannfan) [inline src/SDKs/SqlManagement/Sql.Tests/SessionRecords/Sql.Tests.BlobAuditingTest/TestServerBlobAuditing.json]: See, this is why I'm glad you're looking at this :)

Comment (nathannfan) [inline src/SDKs/SqlManagement/Sql.Tests/SessionRecords/Sql.Tests.BlobAuditingTest/TestServerBlobAuditing.json]: Turns out this was an obsolete recording.  Removed all obsolete recordings.

**Review by markcowl — APPROVED** [review]

**Comment by nathannfan** [comment]
@shahabhijeet , @markcowl  thanks for the quick review, guys!

**Pull request merged**